File: app/src/javax/java/com/adawolfs/mvp/mvp/BaseMvpActivity.java
package com.adawolfs.mvp.mvp;

import android.content.Context;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.adawolfs.mvp.mvp.annotations.Injector;

import javax.inject.Inject;

/**
 * Created by adawolfs on 1/20/17.
 */

public abstract class BaseMvpActivity<V extends BaseMvpView , P extends BaseMvpPresenter<V>>
        extends AppCompatActivity implements BaseMvpView{


    public Context getContext(){
        return this;
    }


    public void showError(String error) {
        Toast.makeText(this, error, Toast.LENGTH_LONG).show();
    }

    public void showError(int stringResId) {
        Toast.makeText(this, stringResId, Toast.LENGTH_LONG).show();
    }

    public void showMessage(int srtResId) {
        Toast.makeText(this, srtResId, Toast.LENGTH_LONG).show();
    }

    public void showMessage(String message) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
    }


}

File: app/src/kotlin/java/com/adawolfs/mvp_java/mvp/BaseMvpActivity.kt
package com.adawolfs.mvp_java.mvp

import android.content.Context
import android.support.v7.app.AppCompatActivity
import org.jetbrains.anko.toast

/**
 * Created by adawolfs on 1/20/17.
 */

abstract class BaseMvpActivity<V : BaseMvpView, P : BaseMvpPresenter<V>> :
        AppCompatActivity(),  BaseMvpView {

    override fun getContext(): Context {
        return this
    }

    override fun showError(error: String) {
        toast(error)
    }

    override fun showError(error: Int) {
        toast(error)
    }

    override fun showMessage(srtResId: Int) {
        toast(srtResId)
    }

    override fun showMessage(message: String) {
        toast(message)
    }

}